interface/main/mobile/m_cal.php

```diff
@@ -0,0 +1,258 @@
+                            $query = "SELECT pc_eid, concat(fname,' ',lname) as name, pc_startTime, pc_hometext, pc_catname, pc_catcolor
+                                    FROM openemr_postcalendar_events as e
+                                    LEFT OUTER JOIN patient_data as p ON e.pc_pid = p.pid
+                                    LEFT JOIN openemr_postcalendar_categories as c ON  e.pc_catid = c.pc_catid
+                                    WHERE pc_eventDate =?";
+                
+                            if (!empty($provider)) {
+                                $query .= " AND e.pc_aid=$provider";
+                            }
+                
+                            $query .= " ORDER BY e.pc_startTime asc";
+                            $result = sqlStatement($query, array($thisyear."-".$thismonth."-".$today));
+                //continue BS4 construct
+                            //if no appointments are found, output a notice
+                            if (sqlNumRows($result) < 1) {
+                                echo "<div class='col-xs-12 alert alert-success'>". xlt('No Appointments Found') ."</div>";
+                            } else {
+                                //echo "<div class='alert alert-success'>".sqlNumRows($result)." ". xlt('Appointments Found') ."</div>";
+                            }
+                            $p_count = 0;
+                            echo "</div>";
+                        while ($row = sqlFetchArray($result)) {
+                            if (is_null($row["name"])) {
+                                echo '<div class="cal_cat" style="background-color:' . $row["pc_catcolor"] . '">';
+                                echo $row["pc_catname"] . " ". $row['pc_hometext'] ."</div>";
+                                continue;
+                            }
+                            echo '<div class="col-xs-12 visit" style="background-color:' . $row["pc_catcolor"] . '">';
+                            echo '<div class="col-xs-4 col-sm-4 col-md-4 col-lg-4">';
+                            echo $row["name"];
+                            echo "</div>";
+                            echo '<div class="col-xs-4 col-sm-4 col-md-4 col-lg-4">' . $row["pc_hometext"] . "</div>";
+                            echo '<div class="col-xs-4 col-sm-4 col-md-4 col-lg-4">' . oeFormatTime($row["pc_startTime"]) . "</div>";
+                    echo "</div>";
+    
+                        }
+                        ?>
+                     </div>
+                     </div>
+            </div>
+        </form>
+    </div>
+
+<?php common_footer($display); ?>
+<script>
+    $(document).ready(function() {
+        $("#prov_select").change(function () {
+            var prov_sched = this.value;
+            top.restoreSession();
+            window.location = "m_cal.php?provider="+prov_sched+"&day=<?php echo $day; ?>&month=<?php echo $month; ?>";
```

Comment:
ok, this is where it gets fun :)
Here, we will use a function that does urlencode followed by js_escape:
```
window.location = "m_cal.php?provider=" + prov_sched + "&day=" + <?php echo js_url($day); ?> + "&month=" + <?php echo js_url($month); ?>;
```
(note how I reworked it here because the js_escape will add quotes to the outside)